akd/agents/search/deep_search.py

```diff
@@ -686,33 +566,19 @@ async def _arun(
                 "content": result.content,
                 "category": getattr(result, "category", "science"),
             }
-            # Preserve relevancy information if available
-            if (
-                hasattr(result, "relevancy_score")
-                and result.relevancy_score is not None
-            ):
-                result_dict["relevancy_score"] = result.relevancy_score
-            if hasattr(result, "should_fetch_full_content"):
-                result_dict["full_content_fetched"] = result.should_fetch_full_content
+            # Preserve any additional metadata from pipeline processing and extra fields
+            if hasattr(result, "extra") and result.extra:
+                result_dict["extra"] = result.extra
             results_as_dicts.append(result_dict)
 
-        # Add research report as the first result
-        if results_as_dicts:
-            results_as_dicts.insert(
-                0,
-                {
-                    "url": "deep-research://report",
-                    "title": "Deep Research Report",
-                    "content": research_output["research_report"],
-                    "category": "research",
-                    "key_findings": research_output["key_findings"],
-                    "quality_score": research_output["evidence_quality_score"],
-                    "iterations": research_output["iterations_performed"],
-                },
-            )
-
         return LitSearchAgentOutputSchema(
             results=results_as_dicts,
```

Comment:
we should probably do `results` as `list[searchresultitem]` itself as well? instead of dict